x/twap/store.go

```diff
@@ -143,7 +133,12 @@ func (k Keeper) getMostRecentRecordStoreRepresentation(ctx sdk.Context, poolId u
 	store := ctx.KVStore(k.storeKey)
 	key := types.FormatMostRecentTWAPKey(poolId, asset0Denom, asset1Denom)
 	bz := store.Get(key)
-	return types.ParseTwapFromBz(bz)
+	twap, err := types.ParseTwapFromBz(bz)
+	if err != nil {
+		err = fmt.Errorf("error in get most recent twap, likely that asset 0 or asset 1 were wrong: %s %s."+
```

Comment:
The only possible error from the callee is proto unmarshalling. What does it mean for them to be wrong?

Reply:
The callee also errors if there is no TWAP found, no?

https://github.com/osmosis-labs/osmosis/blob/c4766c7302de294de902d69c26e192bb4ecfa747/x/twap/types/keys.go#L109-L111

Reply:
yeah, thats the thing we're catching